src/main/java/com/iota/iri/utils/thread/ThreadUtils.java

```diff
@@ -0,0 +1,139 @@
+package com.iota.iri.utils.thread;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+
+/**
+ * This class contains a collection of methods that make the management of {@link Thread}s a little bit more convenient.
+ *
+ * Apart from offering method to spawn and terminate {@link Thread}s, it also offers methods to {@link #sleep(int)}
+ * while correctly maintaining the {@link Thread#isInterrupted()} flags.
+ */
+public class ThreadUtils {
```

Comment:
The synchronized call is not there because of it being a "global" utility class but to prevent the spawning of multiple threads when "start" is being called multiple times. Please note that the synchronization happens over the threadIdentifier, so multiple classes can simultaneously use this class without having to wait for the synchronization.

so if a class implements this functionality:

private ThreadIdentifier myIdentifier = new ThreadIdentifier("Thread Name for console");

...

public void start() {
    ThreadUtils.spawnThread(this::myThreadMethod, myIdentifier);
}

and i call

myClassInstance.start();
myClassInstance.start();
myClassInstance.start(); // from different threads

it still only spawns one thread